components/director/internal/destinationfetchersvc/client.go

```diff
@@ -50,9 +49,16 @@ type DestinationServiceAPIConfig struct {
 
 // Client destination client
 type Client struct {
-	httpClient *http.Client
-	apiConfig  DestinationServiceAPIConfig
-	apiURL     string
+	HTTPClient        *http.Client
+	apiConfig         DestinationServiceAPIConfig
+	authConfig        config.InstanceConfig
+	authToken         string
+	authTokenValidity time.Time
+}
+
+// Close closes idle connections
+func (c *Client) Close() {
+	c.HTTPClient.CloseIdleConnections()
```

Comment:
By default the connection are kept to be reused if possible. 
Would the setting: MaxIdleConnsPerHost work better to prevent leaks

Reply:
The setting is per host. I suspect that we have a relatively small number of hosts that we communicate with.
I don't think it should be 1 either.